File: server/src/main/kotlin/ch/abbts/domain/model/OfferModel.kt
package ch.abbts.domain.model

import kotlinx.serialization.Contextual
import kotlinx.serialization.Serializable
import java.time.LocalDate

@Serializable
data class OfferModel(
    val id: Int? = null,
    val userId: Int,
    val taskId: Int,
    val status: OfferStatus,
    val active: Boolean,
    val text: String,
    val title: String,
    @Contextual val validUntil: LocalDate,
)

File: ui/src/models/OfferModel.ts
export interface Offer {
  id?: number
  title: string
  status?: any
  task?: any
  text: any
  active?: boolean
  validUntil?: string|null
  createdAt?: number
  taskId: number|null
  userId: number|null
}
export interface OfferDto {
  id?: number | null;
  userId: number | null;
  taskId: number | null;
  status?: OfferStatus;
  active?: boolean;
  text: string;
  title?: string | null;
  validUntil?: string | null;
}

export enum OfferStatus {
  SUBMITTED = 'SUBMITTED',
  REJECTED = 'REJECTED',
  ACCEPTED = 'ACCEPTED',
}
